### DISCUSSION TITLE
All output variables

### DISCUSSION THREAD

**Discussion by ashehad** [APSIMInitiative/ApsimX#8549] [General]
Hi there,

Currently, I am running APSIM that has a built up executable and for it, I can look at the output variables using the IntelliSense. However, I would like to look at the output variables in the code. Is there a file that lists all the output variables? If so, could someone point me to github.

Thanks,
Ashehad

**Comment by hol353** [comment]
There is no list of output variables. The variables are scattered across several hundred source code files. Essentially any public property in any c# class that implements IModel in Models.exe is an output variable. Also models are constructed in the user interface without any coding at all using the Plant Modelling Framework) and for these lots of output variables aren't in the source code.

Brown, H.E., Huth, N.I., Holzworth, D.P., Teixeira, E.I., Zyskowski, R.F., Hargreaves, J.N.G., Moot, D.J., 2014. Plant Modelling Framework: Software for building and running crop models on the APSIM platform. Environmental Modelling & Software 62, 385–398. https://doi.org/10.1016/j.envsoft.2014.09.005

**Comment by ashehad** [comment]
Thanks for your quick response and clarifications!

Best,
Ashehad
